snap/hooks/configure

```diff
@@ -126,75 +174,103 @@ done
 # handle usage of database provider
 dbProvider=$(snapctl get dbtype)
 PREV_DB_PROVIDER_FILE="$SNAP_DATA/prevdbtype"
-if [ -f "$PREV_DB_PROVIDER_FILE" ]; then
-    echo "" > "$PREV_DB_PROVIDER_FILE"
-fi
+
 case "$dbProvider" in 
     "")
         # not set, don't do anything 
         ;;
     "redis")
-        # dbtype is set to redis, see what the previous value of dbtype was
         prevDBtype=$(cat "$PREV_DB_PROVIDER_FILE")
         if [ "$prevDBtype" != "redis" ]; then
-            # change from previous database provider to redis
-            # first change the configuration.toml files for the following
-            # services:
+            # change provider to redis:
+            #
+            # first check for service config in consul, if found
+            # update directly, otherwise update configuration.toml
+            # (this should practically only ever happen for the
+            # two support-* services, as they're disabled by
+            # default in the snap).
+            #
+            # * core-command
             # * core-data
             # * core-metadata
             # * support-notifications
             # * support-scheduler
-            for svc in core-data core-metadata support-notifications support-scheduler; do
-                configFile=$SNAP_DATA/config/$svc/res/configuration.toml
-                toml2json --preserve-key-order "$configFile" | \
-                    jq -r '.Databases.Primary.Type = "redisdb" | .Databases.Primary.Port = 6379' | \
-                    json2toml --preserve-key-order > "$configFile.tmp"
-                mv "$configFile.tmp" "$configFile"
-            done
+            for svc in ${DB_SERVICES[*]}; do
+                updated="false"
+
+                exists=`curl -s http://localhost:8500/v1/kv/edgex/core/1.0/edgex-$svc/Databases/Primary/Type?raw`
```

Comment:
this logic doesn't quite work right, when I run this curl command locally I see this as returning the value of the `Type` key, which never ends up being "true" as you check below:

```
$ curl -s http://localhost:8500/v1/kv/edgex/core/1.0/edgex-core-data/Databases/Primary/Type?raw; echo
mongodb
```